plugins/Marketplace/PackageModel.py

```diff
@@ -250,3 +288,99 @@ def isCompatibleMaterialStation(self) -> bool:
     @pyqtProperty(bool, constant = True)
     def isCompatibleAirManager(self) -> bool:
         return self._is_compatible_air_manager
+
+    @pyqtProperty(bool, constant = True)
+    def isBundled(self) -> bool:
+        return self._is_bundled
+
+    @pyqtProperty(str, constant = True)
+    def downloadURL(self) -> str:
+        return self._download_url
+
+    # --- manage buttons signals ---
+
+    stateManageButtonChanged = pyqtSignal()
+
+    installPackageTriggered = pyqtSignal(str, str)
+
+    uninstallPackageTriggered = pyqtSignal(str)
+
+    updatePackageTriggered = pyqtSignal(str, str)
+
+    enablePackageTriggered = pyqtSignal(str)
+
+    disablePackageTriggered = pyqtSignal(str)
+
+    busyChanged = pyqtSignal()
+
+    @pyqtSlot()
+    def install(self):
+        self.setBusy(True)
+        self.installPackageTriggered.emit(self.packageId, self.downloadURL)
+
+    @pyqtSlot()
+    def update(self):
+        self.setBusy(True)
+        self.updatePackageTriggered.emit(self.packageId, self.downloadURL)
+
+    @pyqtSlot()
+    def uninstall(self):
+        self.uninstallPackageTriggered.emit(self.packageId)
+
+    @pyqtProperty(bool, notify= busyChanged)
+    def busy(self):
+        """
+        Property indicating that some kind of upgrade is active.
+        """
+        return self._is_busy
+
+    @pyqtSlot()
+    def enable(self):
+        self.enablePackageTriggered.emit(self.packageId)
+
+    @pyqtSlot()
+    def disable(self):
+        self.disablePackageTriggered.emit(self.packageId)
+
+    def setBusy(self, value: bool):
+        if self._is_busy != value:
+            self._is_busy = value
+            try:
+                self.busyChanged.emit()
+            except RuntimeError:
+                pass
+
+    def _packageInstalled(self, package_id: str) -> None:
+        if self._package_id != package_id:
+            return
+        self.setBusy(False)
+        try:
+            self.stateManageButtonChanged.emit()
+        except RuntimeError:
```

Comment:
The runtime error is caused by the C++ object already being deleted. Setting the ownership to CPP should solve it, but it doesn't.

So we can't move it anywhere else and we really have to listen to the runtime error (unless you know of another way to prevent this issue from occuring)